mathlibtools/file_status.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from typing import Optional, Set
+
+@dataclass(frozen=True)
+class FileStatus:
+  """Capture the file status descriptions in the wiki yaml.
+
+  `string_match` are the substrings that must be found, lowercase match
+  `color` is how the node should be colored if it has the status
+  """
+
+  string_match: Set[str]
+  # colors from X11
+  color: str
+
+  @classmethod
+  def yes(cls) -> "FileStatus":
+    return cls({"yes"}, "green")
+
+  @classmethod
+  def pr(cls) -> "FileStatus":
+    return cls({"no", "pr"}, "lightskyblue")
+
+  @classmethod
+  def wip(cls) -> "FileStatus":
+    return cls({"no", "wip"}, "lightpink")
+
+  @classmethod
+  def no(cls) -> "FileStatus":
+    return cls({"no"}, "orange")
+
+  @classmethod
+  def missing(cls) -> "FileStatus":
+    return cls(set(), "orchid1")
+
```

Comment:
And let's not display this. Most people don't want to know about files being missing from the port-status list. That's for bots to care about, not humans.